packages/local_auth/local_auth_android/lib/local_auth_android.dart

```diff
@@ -0,0 +1,83 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+import 'package:local_auth_platform_interface/local_auth_platform_interface.dart';
+import 'package:local_auth_platform_interface/types/auth_messages.dart';
+import 'package:local_auth_platform_interface/types/auth_options.dart';
+import 'package:local_auth_platform_interface/types/biometric_type.dart';
+
+export 'package:local_auth_android/types/auth_messages_android.dart';
+export 'package:local_auth_platform_interface/types/auth_messages.dart';
+export 'package:local_auth_platform_interface/types/auth_options.dart';
+export 'package:local_auth_platform_interface/types/biometric_type.dart';
+
+const MethodChannel _channel =
+    MethodChannel('plugins.flutter.io/local_auth_android');
+
+/// The implementation of [LocalAuthPlatform] for Android.
+class LocalAuthAndroid extends LocalAuthPlatform {
+  /// Registers this class as the default instance of [LocalAuthPlatform].
+  static void registerWith() {
+    LocalAuthPlatform.instance = LocalAuthAndroid();
+  }
+
+  @override
+  Future<bool> authenticate({
+    required String localizedReason,
+    required Iterable<AuthMessages> authMessages,
+    AuthenticationOptions options = const AuthenticationOptions(),
+  }) async {
+    assert(localizedReason.isNotEmpty);
+    final Map<String, Object> args = <String, Object>{
+      'localizedReason': localizedReason,
+      'useErrorDialogs': options.useErrorDialogs,
+      'stickyAuth': options.stickyAuth,
+      'sensitiveTransaction': options.sensitiveTransaction,
+      'biometricOnly': options.biometricOnly,
+    };
+    for (final AuthMessages messages in authMessages) {
+      args.addAll(messages.args);
```

Comment:
Don't we only want to add the Android messages?

(This would be true for iOS as well; I didn't catch it there.)

Reply:
We want to add any messages that are provided (there is no overlap), but it would be good to add them anyways as defaults in case none are provided.

Reply:
It's not a problem now, but that might indeed solve some issues in case future platforms have overlapping keys. Adding a check.